Packages/com.unity.inputsystem/InputSystem/Plugins/Android/AndroidDevice.cs

```diff
@@ -0,0 +1,63 @@
+#if ((UNITY_EDITOR || UNITY_ANDROID) && UNITY_6000_0_OR_NEWER) || PACKAGE_DOCS_GENERATION
+
+using System;
+using UnityEngine.InputSystem.Controls;
+using UnityEngine.InputSystem.Layouts;
+using UnityEngine.InputSystem.LowLevel;
+
+namespace UnityEngine.InputSystem.Android.LowLevel
+{
+    /// <summary>
+    /// Input device that represents an Android handheld device.
+    /// </summary>
+    /// <remarks> Android device is available starting on 6000.0 and newer. </remarks>
+    [InputControlLayout(displayName = "Android Device", hideInUI = true)]
+    public class AndroidDevice : InputDevice
+    {
+        [InputControl(synthetic = true)]
+        public ButtonControl button { get; protected set; }
```

Comment:
What's this? I see it's a public API, how would user use it ?

Reply:
Not sure how to get rid of this ButtonControl. InputDevices need to have AT LEAST one control. If not they throw an exception. In our case, this controls is not needed

Reply:
```
Could not create a device for 'AndroidDevice (Android)' (exception: System.InvalidOperationException: Control '/AndroidDevice' with layout 'AndroidDevice' has no size set and has no children to compute size from
 at UnityEngine.InputSystem.Layouts.InputDeviceBuilder.ComputeStateLayout (UnityEngine.InputSystem.InputControl control)
 at UnityEngine.InputSystem.Layouts.InputDeviceBuilder.InstantiateLayout (UnityEngine.InputSystem.Layouts.InputControlLayout layout, UnityEngine.InputSystem.Utilities.InternedString variants, UnityEngine.InputSystem.Utilities.InternedString name, UnityEngine.InputSystem.InputControl parent) 
 at UnityEngine.InputSystem.Layouts.InputDeviceBuilder.Setup (UnityEngine.InputSystem.Utilities.InternedString layout, UnityEngine.InputSystem.Utilities.InternedString variants, UnityEngine.InputSystem.Layouts.InputDeviceDescription deviceDescription)
```

Reply:
@ekcoh I have tried pretty much everything but I couldn't find a way to have no `ButtonControl`. This is the exception I am hitting ([code](https://github.com/Unity-Technologies/InputSystem/blob/platform/android/add-back-button-leaves-app-new-input/Packages/com.unity.inputsystem/InputSystem/Devices/InputDeviceBuilder.cs#L766)).

```
Could not create a device for 'AndroidDevice (Android)' (exception: System.InvalidOperationException: Control '/AndroidDevice' with layout 'AndroidDevice' has no size set and has no children to compute size from
 at UnityEngine.InputSystem.Layouts.InputDeviceBuilder.ComputeStateLayout (UnityEngine.InputSystem.InputControl control)
 at UnityEngine.InputSystem.Layouts.InputDeviceBuilder.InstantiateLayout (UnityEngine.InputSystem.Layouts.InputControlLayout layout, UnityEngine.InputSystem.Utilities.InternedString variants, UnityEngine.InputSystem.Utilities.InternedString name, UnityEngine.InputSystem.InputControl parent) 
 at UnityEngine.InputSystem.Layouts.InputDeviceBuilder.Setup (UnityEngine.InputSystem.Utilities.InternedString layout, UnityEngine.InputSystem.Utilities.InternedString variants, UnityEngine.InputSystem.Layouts.InputDeviceDescription deviceDescription)
```